contributor_roadmap.md

```diff
@@ -0,0 +1,165 @@
+# Contributor Roadmap/Expectations
+Hello, thank you for your interest in contributing to Matplotlib! 
+
+Matplotlib is primarily administered via github, so more information on the exact permissions granted to each role is available at [github help](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization#repository-access-for-each-permission-level)
+
+
+# Github
+
+* https://github.com/matplotlib/matplotlib
+
+## Github Organization ownership
+Full control of everything on GH
+**People:**
+- Lead developer, steering council
+- someone from NF (?)
+- someone from outside?
+
+Want to keep this set small enough that we don’t have too much unneeded attack surface area, but big enough that we don’t have single point of failure.
+
+## Administration permissions
+
+Manages the administration of the Matplotlib github repositories. 
+
+* https://github.com/orgs/matplotlib/teams/admin
+
+- Made up of steering council members, appointment process discussed here:
+    - https://github.com/matplotlib/governance/blob/master/governance.md#steering-council
+    - should decouple SC and admin power going forward
+**Responsibilities:**
+- add new repositories and teams to the matplotlib org
+- add members to Matplotlib teams
+- can delete / moderate issues
+    
+- on a GH team that has “Admin” level permissions on all repositories
+## Release Powers
+- has publish permissions to pypi
+- Who has this
+    - Lead developer, steering council, release manager
+- responsibilities
+    - run release process
+
+## Development
+Responsible for the codebase (including documentation)
+
+* https://github.com/orgs/matplotlib/teams/developers
+
+
+**contribute**: https://matplotlib.org/devdocs/devel/index.html
+
+**triage:**
+- triaging issues and pull requests means:
+    - assigning labels, milestones, and reviewers
+    - closing and reopening as needed
+    - marking duplicates
+- get triage privileges: granted on 1st merged pull request
+    - (We need to write the script for this, run as a service on heroku’s free tier?  can we do this with actions?)
+
+**Get commit privileges:**
+Significant or sustained merged pull requests 
+- including domain expertise as needed
+
+**Lose Commit privileges:**
+- no activity on the repository/calls in 6 months
+- repeated or severe violations of [merge guidelines](https://matplotlib.org/devdocs/devel/coding_guide.html)
+- potential repercussion of a [Code of Conduct](https://www.python.org/psf/conduct/) violation
+
+
+# Community / communications
```

Comment:
I like the fact that there is an emphasis on the folks observing the Code of Conduct and that breaches to this would lead to revoking their privileges.  A quick search did not point me to clear guidelines on how to report this or enforce the CoC (maybe my search-fu failed) . 

So maybe it would also be useful to appoint a code of conduct person or subteam

Reply:
On our to-do list is to adopt the Python CoC to an MPL specific one and then include enforcement details and reporting - which will be to NumFocus so that there's an external record of the report.